src/freenas/etc/ix.rc.d/ix-nfsd

```diff
@@ -32,16 +32,36 @@ setgroup()
 	echo "\"${group}\"" | sed -E 's/\\/\\\\/'
 }
 
+get_virtual_fqdn()
+{
+	RO_FREENAS_CONFIG=$(ro_sqlite ${name} 2> /tmp/${name}.fail && rm /tmp/${name}.fail)
+	trap 'rm -f ${RO_FREENAS_CONFIG}' EXIT
+
+	gc_hostname_virtual="$(${FREENAS_SQLITE_CMD} ${RO_FREENAS_CONFIG} \
+		"SELECT gc_hostname_virtual FROM network_globalconfiguration ORDER BY -id LIMIT 1")"
+	gc_domain="$(${FREENAS_SQLITE_CMD} ${RO_FREENAS_CONFIG} \
+		"SELECT gc_domain FROM network_globalconfiguration ORDER BY -id LIMIT 1")"
+
+	if [ -n "${gc_hostname_virtual}" -a -n "${gc_domain}" ]; then
+		echo "${gc_hostname_virtual}.${gc_domain}"
```

Comment:
And if the already put FQDN into `hostname_virtual`, adding another domain to it wouldn't be good.